vignette/Validation.Rmd

```diff
@@ -0,0 +1,155 @@
+---
+title: "Comparison"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Comparison}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>", 
+  fig.path = "man/figures/README-",
+  fig.width = 9, 
+  fig.height = 9
+)
+```
+
+```{r setup}
+library(hypatia)
+library(ggplot2)
+library(tidyverse)
+```
+
+```{r hypatia}
+
+# Create our population
+countryname = "Antigua and Barbuda"
+
+pop <- hypatia:::get_population(countryname)
+
+# Scale it for speed
+pop$n <- round(pop$n/10)
+
+# Create our equivalent parameters
+time_period <- 200
+
+parameters <- get_parameters(
+  population = pop$n,
+  contact_matrix_set = squire::get_mixing_matrix(iso3c = "ATG"),
+  R0 = 2,
+  time_period = time_period,
+  tt_contact_matrix = 0
+)
+
+# run our simulation - default timesteps = 200
+df <- hypatia:::run_simulation(pop, parameters, max_age = 100)
+
+numruns <- length(df)
+numdatapoints <- length(df[[1]][[1]]) - 1
```

Comment:
I think this is meant to be
```suggestion
numdatapoints <- nrow(df) - 1
```

however, as it is we get

```
df[[1]] # the column timestep
df[[1]][[1]] # the first value of that column
length(df[[1]][[1]]) # always 1 because of the second `[[` operation
```

which means the expression always evaluates to zero. However it is unused so can be removed.